models/Note.js

```diff
@@ -84,23 +85,34 @@ class Note extends BaseModel {
     reader /*: any */,
     note /*: any */
   ) /*: Promise<any> */ {
-    const props = _.pick(note, [
-      'noteType',
-      'content',
-      'selector',
-      'json',
-      'documentId',
-      'publicationId'
-    ])
+    const { Document } = require('./Document')
+
+    const props = _.pick(note, ['noteType', 'content', 'selector', 'json'])
 
+    // $FlowFixMe
+    const path = urlparse(note.inReplyTo).path.substr(45)
+    props.selector = note['oa:hasSelector']
```

Comment:
At some point we could clean this up by using our own context where 'oa:hasSelector' is aliased to 'selector' and we won't have to do this reassignment runaround thing anymore. 🙂